python/tchannel/tornado/peer.py

```diff
@@ -0,0 +1,448 @@
+# Copyright (c) 2015 Uber Technologies, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+import logging
+from itertools import chain
+from random import random
+from collections import deque
+
+from tornado import gen
+
+
+try:
+    # included in Tornado 4.2
+    from tornado.locks import Condition
+except ImportError:  # pragma: no cover
+    from toro import Condition
+
+from .dispatch import Request
+from .stream import InMemStream, Stream
+from .connection import StreamConnection
+from ..handler import CallableRequestHandler
+
+
+log = logging.getLogger('tchannel')
+
+
+class PeerManager(object):
```

Comment:
Please don't end classes in `Manager` - in my experience this always leads to bucket classes that grow far beyond their original intention (`UserManager` gives me nightmares, for example). Instead, lets choose a name that is more specific. 

Suggest renaming to `PeerGroup`.